hamilton/plugins/pandas_extensions.py

```diff
@@ -983,6 +993,165 @@ def load_data(self, type: Type) -> Tuple[DATAFRAME_TYPE, Dict[str, Any]]:
+        self.escapechar = escapechar
+        self.comment = comment
+        self.encoding = encoding
+        self.dialect = dialect
+        self.error_bad_lines = error_bad_lines
+        self.warn_bad_lines = warn_bad_lines
+        self.delim_whitespace = delim_whitespace
+        self.low_memory = low_memory
+        self.memory_map = memory_map
+        self.float_precision = float_precision
+        self.storage_options = storage_options
+
+    @classmethod
+    def applicable_types(cls) -> Collection[Type]:
+        return [DataFrame]
+
+    def _get_loading_kwargs(self) -> Dict[str, Any]:
+        kwargs = {
+            "sep": self.sep,
+            "delimiter": self.delimiter,
+            "header": self.header,
+            "names": self.names,
+            "index_col": self.index_col,
+            "usecols": self.usecols,
+            "engine": self.engine,
+            "converters": self.converters,
+            "true_values": self.true_values,
+            "false_values": self.false_values,
+            "skipinitialspace": self.skipinitialspace,
+            "skiprows": self.skiprows,
+            "skipfooter": self.skipfooter,
+            "nrows": self.nrows,
+            "na_values": self.na_values,
+            "keep_default_na": self.keep_default_na,
+            "na_filter": self.na_filter,
+            "verbose": self.verbose,
+            "skip_blank_lines": self.skip_blank_lines,
+            "parse_dates": self.parse_dates,
+            "infer_datetime_format": self.infer_datetime_format,
+            "keep_date_col": self.keep_date_col,
+            "date_parser": self.date_parser,
+            "dayfirst": self.dayfirst,
+            "cache_dates": self.cache_dates,
+            "iterator": self.iterator,
+            "chunksize": self.chunksize,
+            "compression": self.compression,
+            "thousands": self.thousands,
+            "decimal": self.decimal,
+            "lineterminator": self.lineterminator,
+            "quotechar": self.quotechar,
+            "quoting": self.quoting,
+            "doublequote": self.doublequote,
+            "escapechar": self.escapechar,
+            "comment": self.comment,
+            "encoding": self.encoding,
+            "dialect": self.dialect,
+            "error_bad_lines": self.error_bad_lines,
+            "warn_bad_lines": self.warn_bad_lines,
+            "delim_whitespace": self.delim_whitespace,
+            "low_memory": self.low_memory,
+            "memory_map": self.memory_map,
+            "float_precision": self.float_precision,
+            "storage_options": self.storage_options,
+        }
+        return kwargs
+
+    def load_data(self, type: Type) -> Tuple[DataFrame, Dict[str, Any]]:
+        filepath_or_buffer = get_filepath_or_buffer(
+            self.path_or_buffer
+        )
+        kwargs = self._get_loading_kwargs()
+        data = read_table(filepath_or_buffer, **kwargs)
+        return data, {}
```

Comment:
this should use the metadata utility like the others are using to provide metadata on the file being loaded.